token-lending/cli/src/main.rs

```diff
@@ -0,0 +1,700 @@
+    let user_collateral_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+    let liquidity_supply_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+    let liquidity_fee_receiver_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+    let liquidity_host_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(Token::LEN)?;
+
+    let total_balance = reserve_balance
+        + collateral_mint_balance
+        + collateral_supply_balance
+        + user_collateral_balance
+        + liquidity_supply_balance
+        + liquidity_fee_receiver_balance
+        + liquidity_host_balance;
+
+    let mut create_accounts_transaction_1 = Transaction::new_with_payer(
+        &[
+            create_account(
+                &config.fee_payer.pubkey(),
+                &reserve_keypair.pubkey(),
+                reserve_balance,
+                Reserve::LEN as u64,
+                &spl_token_lending::id(),
+            ),
+            create_account(
+                &config.fee_payer.pubkey(),
+                &collateral_mint_keypair.pubkey(),
+                collateral_mint_balance,
+                Mint::LEN as u64,
+                &spl_token::id(),
+            ),
+            create_account(
+                &config.fee_payer.pubkey(),
+                &collateral_supply_keypair.pubkey(),
+                collateral_supply_balance,
+                Token::LEN as u64,
+                &spl_token::id(),
+            ),
+            create_account(
+                &config.fee_payer.pubkey(),
+                &user_collateral_keypair.pubkey(),
+                user_collateral_balance,
+                Token::LEN as u64,
+                &spl_token::id(),
+            ),
+        ],
+        Some(&config.fee_payer.pubkey()),
+    );
+
+    let mut create_accounts_transaction_2 = Transaction::new_with_payer(
+        &[
+            create_account(
+                &config.fee_payer.pubkey(),
+                &liquidity_supply_keypair.pubkey(),
+                liquidity_supply_balance,
+                Token::LEN as u64,
+                &spl_token::id(),
+            ),
+            create_account(
+                &config.fee_payer.pubkey(),
+                &liquidity_fee_receiver_keypair.pubkey(),
+                liquidity_fee_receiver_balance,
+                Token::LEN as u64,
+                &spl_token::id(),
+            ),
+            create_account(
+                &config.fee_payer.pubkey(),
+                &liquidity_host_keypair.pubkey(),
+                liquidity_host_balance,
+                Token::LEN as u64,
+                &spl_token::id(),
+            ),
+        ],
+        Some(&config.fee_payer.pubkey()),
+    );
+
+    let mut init_reserve_transaction = Transaction::new_with_payer(
+        &[
+            approve(
```

Comment:
Every approve instruction should have a corresponding revoke instruction at the end of the transaction in case some of the delegated amount wasn't transferred

Reply:
Done.